### DISCUSSION TITLE
shrink image before upload

### DISCUSSION THREAD

**Discussion by aniltirola** [livewire/livewire#4313] [1. Help] [answered]
I use file uploads according to https://laravel-livewire.com/docs/2.x/file-uploads#basic-upload;

I would like to shrink images with javascript, before livewire uploads it to the server. Shrinking is made on the clientside with the javascript-api canvas - `drawImage`. `drawImage` gives me a base64-encoded image.
How do i pass this correctly to livewire? 
Is there a way to tell livewire "Use this base64-data instead of the value from the file-input-field"?

**Comment by jonneroelofs** [comment, marked as answer]
You can upload a file with javascript instead of using wire:model on an input:
https://laravel-livewire.com/docs/2.x/file-uploads#js-api
